### PR TITLE
Limit requesting prior tasks for ranking in scheduler

### PR DESCRIPTION
**Pull request by jpbruinsslot**
### Changes

Limit requesting prior task in the `BoefjeScheduler` for ranking. At the moment we don't leverage ranking of tasks based on prior task runs. At least we should limit the number of tasks. This PR will limit this to the last task that has run, since the ranker only uses the last run task at the moment.

### QA notes

Nothing really to QA here. We limit the number of tasks we retrieve from the database to make ranking a bit more optimized. Regular running of scanning jobs should continue to work. Mainly rescanning a job that has already been executed can be tested to make sure that it continues to work.

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: mula/scheduler/rankers/boefje.py [hunk 1]
```diff
@@ -31,11 +31,11 @@ def rank(self, obj: Any) -> int:
         grace_period = timedelta(seconds=self.ctx.config.pq_grace_period)
 
         # New tasks that have not yet run before
-        if obj.prior_tasks is None or not obj.prior_tasks:
+        if obj.latest_task is None or not obj.latest_task:
```
Comment (ammar92) [inline mula/scheduler/rankers/boefje.py:34]: Just to confirm: can the `latest_task` property ever be a falsy value other than `None`?

**Review by ammar92 — APPROVED** [review]
Nice work. Tiny question, looks good otherwise

File: mula/scheduler/rankers/boefje.py (hunk 1 above)
Comment (underdarknl) [inline mula/scheduler/rankers/boefje.py:34]: What exactly is 'obj' in this scope anyway? Is it the OOI or the to be scheduled job?

Comment (jpbruinsslot) [reply mula/scheduler/rankers/boefje.py:34]: yes it would be possible to be none if there aren't any prior tasks available

Comment (jpbruinsslot) [reply mula/scheduler/rankers/boefje.py:34]: https://github.com/minvws/nl-kat-coordination/blob/e064fe953c04d93176b4ce7784ff7339b03c2eab/mula/scheduler/schedulers/boefje.py#L566

At this moment a `SimpleNamespace`. I need to review how and what we want to be doing for ranking in the future.

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. Onboarding works, rescheduling of tasks works and reports can also be (re)scheduled. 

### What doesn't work:
N/a

### Bug or feature?:
N/a

**Pull request merged**